### ISSUE TITLE
Sign issues

### ISSUE THREAD

**Issue by BluXDragon** [FallenMoonNetwork/CanaryLib#55]
Wall signs (all four directions) return null for "block.getType()"

And casting to Sign;

```
Sign sign = (Sign)block;
```

for ground signs throws an exception

```
2013-04-30 17:39:36 [WARNING] [CanaryMod] Exception while executing Hook: BlockRightClickHook in PluginListener: SignCommandsListener (Plugin: SignCommands)
net.canarymod.hook.HookExecutionException
        at net.canarymod.hook.HookExecutor$1.execute(HookExecutor.java:68)
        at net.canarymod.plugin.RegisteredPluginListener.execute(RegisteredPluginListener.java:55)
        at net.canarymod.hook.HookExecutor.callHook(HookExecutor.java:111)
        at net.minecraft.server.NetServerHandler.a(NetServerHandler.java:457)
        at net.minecraft.server.Packet15Place.a(SourceFile:58)
        at net.minecraft.server.TcpConnection.b(SourceFile:350)
        at net.minecraft.server.NetServerHandler.d(NetServerHandler.java:72)
        at net.minecraft.server.NetworkListenThread.b(SourceFile:35)
        at net.minecraft.server.DedicatedServerListenThread.b(SourceFile:30)
        at net.minecraft.server.MinecraftServer.r(MinecraftServer.java:475)
        at net.minecraft.server.DedicatedServer.r(DedicatedServer.java:206)
        at net.minecraft.server.MinecraftServer.q(MinecraftServer.java:390)
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:329)
        at net.minecraft.server.ThreadMinecraftServer.run(SourceFile:573)
Caused by: java.lang.ClassCastException: net.canarymod.api.world.blocks.CanaryBlock cannot be cast to net.canarymod.api.world.blocks.Sign
        at com.bluxdragon.signcommands.SignCommandsListener.onBlockRightClick(SignCommandsListener.java:60)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at net.canarymod.hook.HookExecutor$1.execute(HookExecutor.java:62)
        ... 13 more
```

**Comment by WWOL** [comment]
Should Sign sign = (Sign)block; work? A sign is a complex block.  Try getting a complex block at that location and casting to sign.

WWOL

**Comment by BluXDragon** [comment]
there is no Block .getComplexBlock method though, so it would essentially be

```
ComplexBlock cb = (ComplexBlock)block;
Sign s = (Sign)cb;
```

 which in the end is the same thing, no?

(Sign)block; is what I was told to use in IRC, in any case.

**Comment by ayyron-dev** [comment]
Complex block doesn't extend block I'm pretty sure. I think the method to get it is in world. Either way this probably needs a looking at to makesure it is the best it can be or needs an improvement.

**Comment by darkdiplomat** [comment]
Complex blocks don't extend block, if you want the block it would be Sign.getBlock()  I think

**Comment by darkdiplomat** [comment]
fixed with 2228e5d  Someone forgot to add the name to Wall Sign so it wasn't getting registered.
